stream/batch.js

```diff
@@ -9,6 +9,9 @@ const streamFactory = (options) => {
   const streamOptions = _.pick(options, 'concurrency')
   const extraFields = _.get(options, 'fields')
 
+  // merge base fields with any user-specified fields
+  const fields = _.assign({}, baseFields, extraFields)
```

Comment:
this change is unrelated, I noticed during the rebase that it's not necessary to call for every row of the CSV file.